evmcore/blockchain_test.go

```diff
@@ -0,0 +1,134 @@
+package evmcore
+
+import (
+	"bytes"
+	"fmt"
+	"math/big"
+	"testing"
+
+	"github.com/unicornultrafoundation/go-u2u/common"
+	"github.com/unicornultrafoundation/go-u2u/core/rawdb"
+	"github.com/unicornultrafoundation/go-u2u/core/state"
+	"github.com/unicornultrafoundation/go-u2u/core/types"
+	"github.com/unicornultrafoundation/go-u2u/core/vm"
+	"github.com/unicornultrafoundation/go-u2u/crypto"
+	"github.com/unicornultrafoundation/go-u2u/params"
+)
+
+var (
+	// A sender who makes transactions, has some funds
+	key, _  = crypto.HexToECDSA("b71c71a67e1177ad4e901695e1b4b9ee17ae16c6668d313eac2f96dbcda3f291")
+	address = crypto.PubkeyToAddress(key.PublicKey)
+	funds   = big.NewInt(1_000_000_000_000_000_000)
+
+	signer = types.HomesteadSigner{}
+)
+
+func GetTx(nonce *uint64, to *common.Address, data []byte) *types.Transaction {
+	tx, _ := types.SignNewTx(key, signer, &types.LegacyTx{
+		Nonce:    *nonce,
+		GasPrice: big.NewInt(1),
+		Gas:      100_000,
+		To:       to,
+		Data:     data,
+	})
+	*nonce++
+	return tx
+}
+
+func GenerateChainWithGenesis(gen func(int, *BlockGen)) ([]*EvmBlock, []types.Receipts, DummyChain) {
+	var (
+		n          = 4
+		config     = params.AllProtocolChanges
+		db         = rawdb.NewMemoryDatabase()
+		statedb, _ = state.New(common.Hash{}, state.NewDatabase(db), nil)
+	)
+	config.EIP712Block = big.NewInt(2)
+	return GenerateChain(config, MustApplyFakeGenesis(statedb, FakeGenesisTime, map[common.Address]*big.Int{
+		address: funds,
+		//dest: types.GenesisAlloc{ Balance: 0, Code: code }
+	}), db, n, gen)
+}
+
+func TestEVMTransition(t *testing.T) {
```

Comment:
Need another transient storage reset test like https://github.com/ethereum/go-ethereum/blob/8032b63f167938e6441d4be3eb49b12014d9bfc8/core/blockchain_test.go#L4029

Reply:
I reckon that the TSTORE logic is simple enough that more integration tests are needed instead

Reply:
Sure we need more integration tests, but the priciple of unit testing is trying to cover more cases. These two method of testing should take different repsonsibilities. So we need to cover as much cases we found out for more code coveraged, as well as detect clashing with future changes